src/util/rocksdb/event_listener.rs

```diff
@@ -65,6 +65,13 @@ impl rocksdb::EventListener for EventListener {
         STORE_ENGINE_COMPACTION_NUM_CORRUPT_KEYS_VEC
             .with_label_values(&[&self.db_name, info.cf_name()])
             .inc_by(info.num_corrupt_keys() as i64);
+        STORE_ENGINE_COMPACTION_REASON_VEC
+            .with_label_values(&[
+                &self.db_name,
+                info.cf_name(),
+                &info.compaction_reason().to_string(),
```

Comment:
em, need `to_string` here?

Reply:
CompactionReason is a enum type which implement fmt::Display, so need to_string. And a compaction job only call this function once, so it is not frequency operation.